### PR TITLE
R2 Final Review

### PR DESCRIPTION
**Pull request by ServiceDog**

### CODE REVIEW

File: server/src/rest/server.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        .route("/cargo/vertiports", routing::post(api::query_vertiports))
+        .layer(Extension(grpc_clients)); // Extension layer must be last
+
+    let address = format!("[::]:{rest_port}").parse().unwrap();
+    rest_info!("(rest) hosted at {:?}", address);
+    axum::Server::bind(&address)
+        .serve(app.into_make_service())
+        .with_graceful_shutdown(shutdown_signal("rest"))
+        .await
+        .unwrap();
```
Comment (GoodluckH) [inline server/src/rest/server.rs:31]: handle unwrap

File: client-rest/examples/rest.rs [hunk 2]
```diff
@@ -61,9 +62,10 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
...
     {
-        let data = FlightConfirm {
+        let data = ItineraryConfirm {
             // Arbitrary UUID
-            fp_id: "cabcdd14-03ab-4ac0-b58c-dd4175bc587e".to_string(),
+            id: uuid::Uuid::new_v4().to_string(),
+            user_id: uuid::Uuid::new_v4().to_string(),
         };
         let data_str = serde_json::to_string(&data).unwrap();
         let uri = format!("{}/cargo/confirm", url);
```
Comment (GoodluckH) [inline client-rest/examples/rest.rs:71]: handle unwrap

File: client-rest/examples/rest.rs [hunk 3]
```diff
@@ -83,9 +85,9 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
 
     // DELETE /cargo/cancel
     {
-        let data = FlightCancel {
+        let data = ItineraryCancel {
             // arbitrary UUID
-            fp_id: "cabcdd14-03ab-4ac0-b58c-dd4175bc587e".to_string(),
+            id: "cabcdd14-03ab-4ac0-b58c-dd4175bc587e".to_string(),
         };
         let data_str = serde_json::to_string(&data).unwrap();
```
Comment (GoodluckH) [inline client-rest/examples/rest.rs:92]: handle unwrap

File: client-rest/examples/rest.rs [hunk 4]
```diff
@@ -105,15 +107,16 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
...
-            timestamp_arrive_min: None,
-            timestamp_arrive_max: None,
+            time_depart_window: Some(TimeWindow {
+                timestamp_min: depart_timestamp_min,
+                timestamp_max: depart_timestamp_min + Duration::seconds(360),
+            }),
+            time_arrive_window: None,
             cargo_weight_kg: 1.0,
         };
         let data_str = serde_json::to_string(&data).unwrap();
```
Comment (GoodluckH) [inline client-rest/examples/rest.rs:122]: handle unwrap

File: server/src/grpc/server.rs [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+pub struct ServiceImpl {}
+
+// Implementing gRPC interfaces for this microservice
+#[tonic::async_trait]
+impl RpcService for ServiceImpl {
+    /// Replies true if this server is ready to serve others.
+    /// # Arguments
+    /// * `request` - the query object with no arguments
+    /// # Returns
+    /// * `ReadyResponse` - Returns true
```
Comment (GoodluckH) [inline server/src/grpc/server.rs:23]:
```suggestion
```

Comment (ServiceDog) [reply server/src/grpc/server.rs]: https://github.com/Arrow-air/svc-cargo/pull/34/commits/cbe42d09bf0f5c1de77460d89a34f66cff254cae

File: client-rest/examples/rest.rs (hunk 4 above)
Comment (ServiceDog) [reply client-rest/examples/rest.rs]: https://github.com/Arrow-air/svc-cargo/pull/34/commits/cbe42d09bf0f5c1de77460d89a34f66cff254cae

File: client-rest/examples/rest.rs (hunk 3 above)
Comment (ServiceDog) [reply client-rest/examples/rest.rs]: https://github.com/Arrow-air/svc-cargo/pull/34/commits/cbe42d09bf0f5c1de77460d89a34f66cff254cae

File: client-rest/examples/rest.rs (hunk 2 above)
Comment (ServiceDog) [reply client-rest/examples/rest.rs]: https://github.com/Arrow-air/svc-cargo/pull/34/commits/cbe42d09bf0f5c1de77460d89a34f66cff254cae

File: server/src/grpc/server.rs [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+        .set_serving::<RpcServiceServer<ServiceImpl>>()
+        .await;
+
+    grpc_info!("(grpc) hosted at {}", addr);
+    tonic::transport::Server::builder()
+        .add_service(health_service)
+        .add_service(RpcServiceServer::new(imp))
+        .serve_with_shutdown(addr, shutdown_signal("grpc"))
+        .await
+        .unwrap();
```
[the remark this answers is not in the document]
Comment (ServiceDog) [reply server/src/grpc/server.rs]: https://github.com/Arrow-air/svc-cargo/pull/34/commits/cbe42d09bf0f5c1de77460d89a34f66cff254cae

**Review by GoodluckH — APPROVED** [review]

**Pull request merged**